mockwebserverwrapper/src/test/java/okhttp3/mockwebserver/KotlinSourceModernTest.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.mockwebserver
+
+import okhttp3.Handshake
+import okhttp3.Headers
+import okhttp3.Headers.Companion.headersOf
+import okhttp3.HttpUrl
+import okhttp3.Protocol
+import okhttp3.TlsVersion
+import okhttp3.WebSocketListener
+import okhttp3.internal.http2.Settings
+import okio.Buffer
+import org.junit.Ignore
+import org.junit.Test
+import java.net.InetAddress
+import java.net.Proxy
+import java.net.Socket
+import java.util.concurrent.TimeUnit
+import javax.net.ServerSocketFactory
+import javax.net.ssl.SSLSocketFactory
+
+/**
+ * Access every type, function, and property from Kotlin to defend against unexpected regressions in
+ * modern 4.0.x kotlin source-compatibility.
+ */
+@Suppress(
+    "ASSIGNED_BUT_NEVER_ACCESSED_VARIABLE",
+    "UNUSED_ANONYMOUS_PARAMETER",
+    "UNUSED_VALUE",
+    "UNUSED_VARIABLE",
+    "VARIABLE_WITH_REDUNDANT_INITIALIZER",
+    "RedundantLambdaArrow",
+    "RedundantExplicitType",
+    "IMPLICIT_NOTHING_AS_TYPE_PARAMETER"
+)
+class KotlinSourceModernTest {
```

Comment:
This is a copy-paste of the other KotlinSourceModernTest, with these changes:
 - uses the obsolete MockWebServer package (okhttp3.mockwebserver)
 - doesn't subtype QueueDispatcher